src/EventStore.Core.Tests/Services/Transport/Http/build_requested_url_should.cs

```diff
@@ -52,43 +52,43 @@ public void with_advertise_ip_and_http_port_set_both_host_and_port_is_changed()
         }
 
         [Test]
-        public void with_port_forward_header_only_port_is_changed()
+        public void with_port_forward_header_only_nothing_is_changed()
```

Comment:
Yes, there's a small change which is quite difficult to notice :) The port in the Assert.AreEqual was 4321 and I've changed it to 1234.

I've modified the behaviour of BuildRequestedUrl to change the base path only when the 3 headers: X-Forwarded-Scheme, X-Forwarded-Host and X-Forwarded-Port are all present. This is to ensure that the correct URL is being rebuilt. I saw some cases where reverse proxies are adding the Host header but not the Port or scheme which might not work properly. In any case, the 3 headers are actually no longer required since URLs will be rewritten as relative URLs. But the functionality is still supported in case someone wants it.